pumps/mongo_aggregate_test.go

```diff
@@ -125,3 +125,113 @@ func TestDoAggregatedWritingWithIgnoredAggregations(t *testing.T) {
 		})
 	}
 }
+
+func TestAggregationTime(t *testing.T) {
+	cfgPump1 := make(map[string]interface{})
+	cfgPump1["mongo_url"] = "mongodb://localhost:27017/tyk_analytics"
+	cfgPump1["ignore_aggregations"] = []string{"apikeys"}
+	cfgPump1["use_mixed_collection"] = true
+
+	pmp1 := MongoAggregatePump{}
+
+	timeNow := time.Now()
+	keys := make([]interface{}, 1)
+	keys[0] = analytics.AnalyticsRecord{APIID: "api1", OrgID: "123", TimeStamp: timeNow, APIKey: "apikey1"}
+
+	tests := []struct {
+		testName              string
+		AggregationTime       int
+		WantedNumberOfRecords int
+	}{
+		{
+			testName:              "create record every 60 minutes - 180 minutes hitting the API",
+			AggregationTime:       60,
+			WantedNumberOfRecords: 3,
+		},
+		{
+			testName:              "create new record every 30 minutes - 120 minutes hitting the API",
+			AggregationTime:       30,
+			WantedNumberOfRecords: 4,
+		},
+		{
+			testName:              "create new record every 15 minutes - 90 minutes hitting the API",
+			AggregationTime:       15,
+			WantedNumberOfRecords: 6,
+		},
+		{
+			testName:              "create new record every 7 minutes - 28 minutes hitting the API",
+			AggregationTime:       7,
+			WantedNumberOfRecords: 4,
+		},
+		{
+			testName:              "create new record every 3 minutes - 24 minutes hitting the API",
+			AggregationTime:       3,
+			WantedNumberOfRecords: 8,
+		},
+		{
+			testName:              "create new record every minute - 10 minutes hitting the API",
+			AggregationTime:       1,
+			WantedNumberOfRecords: 10,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.testName, func(t *testing.T) {
+			cfgPump1["aggregation_time"] = test.AggregationTime
+			errInit1 := pmp1.Init(cfgPump1)
+			if errInit1 != nil {
+				t.Error(errInit1)
+				return
+			}
+
+			defer func() {
+				// we clean the db after we finish every test case
+				sess := pmp1.dbSession.Copy()
+				defer sess.Close()
+
+				if err := sess.DB("").DropDatabase(); err != nil {
+					panic(err)
+				}
+			}()
+
+			ctx := context.TODO()
+			for i := 0; i < test.WantedNumberOfRecords; i++ {
+				for index := 0; index < test.AggregationTime; index++ {
+					errWrite := pmp1.WriteData(ctx, keys)
+					if errWrite != nil {
+						t.Fatal("Mongo Aggregate Pump couldn't write records with err:", errWrite)
+					}
+				}
+				timeNow = timeNow.Add(time.Minute * time.Duration(test.AggregationTime))
+				keys[0] = analytics.AnalyticsRecord{APIID: "api1", OrgID: "123", TimeStamp: timeNow, APIKey: "apikey1"}
+			}
+
+			collectionName := analytics.AgggregateMixedCollectionName
+
+			thisSession := pmp1.dbSession.Copy()
+			defer thisSession.Close()
+
+			analyticsCollection := thisSession.DB("").C(collectionName)
+
+			// we build the query using the timestamp as we do in aggregated analytics
```

Comment:
@mativm02 you are not using the timestamp in the query 😛

Reply:
I forgot to modify the comment